libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -537,31 +540,24 @@ method start*(g: GossipSub) {.async.} =
   ## start pubsub
   ## start long running/repeating procedures
 
-  # interlock start to to avoid overlapping to stops
-  await g.heartbeatLock.acquire()
-
-  # setup the heartbeat interval
-  g.heartbeatRunning = true
-  g.heartbeatFut = g.heartbeat()
-
-  g.heartbeatLock.release()
+  withLock g.heartbeatLock:
```

Comment:
so the heartbeat lock is taken both inside and outside `g.heartbeat`? how does that work?

Reply:
We probably don't want this one here, some mixup with @sinkingsugar refactors that I picked along the way.

Reply:
But it's released every iteration before sleep in heartbeat no?

Reply:
But yeah could be mixup

Reply:
Ah, it is actually correct after all, both this one and the one in `stop`.